### DISCUSSION TITLE
In the quest for a smarter robotic cleaning routine

### DISCUSSION THREAD

**Discussion by oangelo** [Hypfer/Valetudo#1686] [Q&A / Support]
Valetudo is a great tool with a great web interface. In the manual, you can find how to integrate Valetudo with the Home Assistant to have another interface to control the robot and much more.
However, I can't stop wondering how we can use Home Assistant to do “much more”. So I think it would be nice to have an automation suggestion list, with ideas to improve the robot's usability beyond the original software.

This blog has some nice suggestions: https://leonardosmarthomemakers.com/how-to-automate-a-vacuum-robot-in-home-assistant/

1- The automation I think is most useful is to implement a “Reminder  to clean the robot bin”:
Since there is no sensor to identify when the bin is full, a great idea is to clean the bin after a few vacuuns. This can be done by creating a counter in Home Assistant and incrementing it with an automation when the robot “starts cleaning”, and the counter can be reseted when “dust_bin_attachment” is “turned_off”. Now, just set an automation to notify you when the counter is above 2 or 3, depending on what works for you best.

2- I also have automations to return the robot to the charging station automatically as children or pets may unintentionally remove it from the charging station.

3 - I also implemented notifications on the days before robot vacuuming to remind me to look around the rooms for potential objects that might get in the way of the robot cleaning.

What else useful automations you guys are using out there?

**Comment by NotExpectedYet** [comment]
I've extended my smarness a little bit with home assistant. 

1 - Clean rooms based on activity threshold: I use a counter variable that tracks how many movement's happened in a specific room. I'm using this to guage how "active" a room has been, the hoover will only go clean the most active area's which means it's running a hell of a lot less than it use to do with basic schedules. It cleans when it needs pretty much. It's a bit of trial and error to setup, I just used 1 motion event for 1 active count in the room. Some rooms we're way more active than others and needed different thresholds. 

2 - Battery Saving: Mine will use a smart plug to turn on the base station if it get's towards 20 percent whilst docked to start it charging. This also then monitors for when the battery hits over 80% and turns off the base station power. This does have an unintended consequence that the base station might not be available when it's going home, that's solved with a simple trigger on the "Returning to Dock" state and a flick of the base station back on. Infact, I think I did that by just making sure the base station is on when it's cleaning. 

3 - Daily cleaning schedule notification: Because I dunno what rooms are needing to be cleaned, I get a little notification each morning only when a clean is planned for that day. Then I get a completed notification with the rooms it's just done. 

Other ideas: 

I've got a few other ideas, like warning me if a door is closed when a clean is planned. That can't reach area warning seems to stop an entire clean on mine when it crops up. I wanna make sure the doors are available and only clean that room if it is indeed available. 

Honestly not sure what else...

**Comment by oangelo** [comment]
Great idea @NotExpectedYet! I think your concept of "Clean rooms based on activity threshold" is brilliant. Do you happen to have the code available? It would be helpful to have a blueprint to work from.

However, I'm not quite sure I understand the purpose of the "battery saving" automation. Does this feature help prolong the life of the battery? I was under the impression that the robot would already have internal controls for managing its power usage.

**Reply by NotExpectedYet** [reply]
> Great idea @NotExpectedYet! I think your concept of "Clean rooms based on activity threshold" is brilliant. Do you happen to have the code available? It would be helpful to have a blueprint to work from.

Thank you! I'd not seen the idea anywhere else online so thought it was worth mentioning. It works really well! I don't have a blueprint and doubt I'll get time to make one as there's quite a lot of interconnecting parts. I will explain how I did it though, it's not too hard. 

1. Each room needs a variable for a count, I customised this so I could link it to my room id's from Valetudo following their documentation.
2. Another variable to hold the rooms threshold, customised the same as Number 1. 
3. One automation that counts up that variable everytime a motion trigger is triggered for the relevant PIR sensor or whatever your using. 
4. I then have a daily schedule run that firstly checks if the room has a value above the threshold varaible. That fires off a script that will start my hoover going. I use the same breakout method as the valetudo documentation for pulling out the varibales for the room id and how many movements have occured.
5. That script awaits for the hoover to finish the run and then resets any variables to 0 only if that room got hoovered. 

I've just moved house so I don't have a PC setup, I'll dig out all of my .yaml files when I get that setup for you. 

> However, I'm not quite sure I understand the purpose of the "battery saving" automation. Does this feature help prolong the life of the battery? I was under the impression that the robot would already have internal controls for managing its power usage.

Sorry, battery saving isn't that great desciption. It's a bit of power saving and battery life improving. As far as I'm aware, majority of manufacture's I've gotten devices from don't try to prolong the life of your battery. From what I've looked up they last much longer when kept between the 20-80 range. So I originally realised pretty quickly that I'm now wasting power keeping my hoover fully topped up as it's running a lot less thant it did without the automation. The 20-80 thing keeps my battery just topped up enough to not waste power and hopefully prolong it's life if it's indeed correct about the battery life extension. 

Hope that helps!
